app/models/setting.rb

```diff
@@ -272,7 +272,12 @@ def self.load_defaults
     true
   end
 
-  def self.set(name, description, default, full_name = nil, value = nil)
+  def self.set(name, description, default, full_name = nil, value = nil, options = {})
+    if options.has_key? :collection
+      SettingsHelper.module_eval do
+        define_method("#{name}_collection".to_sym){ options[:collection].call }
```

Comment:
Thanks, this is much better. I wonder why passing `Procs` and calling them here instead of just passing the `Hash` directly in `:collection`? It seems like an easier API to me, like this:

`{ :collection => { Hash[Organization.all.map{ |org| [org[:title], org[:title]] } ] } }`